### PR TITLE
fix(node:http/https): fix passing `URL` objs to `http.request`(#2253)

### LINKED ISSUE #2253: Node https interop bug with https.request to fetch mapping
### What version of Bun is running?

0.5.7

### What platform is your computer?

Darwin 22.3.0 arm64 arm

### What steps can reproduce the bug?

```javascript
import * as https from "node:https";

const body = `{ "aps" : { "alert" : "Hello" } }`;
const method = "POST";

const headers = {
  "Authorization": "",
  "x-ms-version": API_VERSION,
  "Content-Type": "application/json",
  "apns-priority": "10",
  "apns-push-type": "alert",
};

async function sendRequest(url, headers, body, method) {
  const options = {
    headers,
    method
  };

  return new Promise((resolve, reject) => {
    const request = https.request(url, options, (response) => {
      const res = {
        body: "",
        headers: response.headers
      }

      response.on("data", (data) => {
        res.body += data;
      });

      response.on("end", () => {
        resolve(res);
      });
    });

    request.on("error", (err) => {
      reject(err);
    });

    request.write(body);
    request.end();
  });
}

const url = new URL("https://some-fake-address.servicebus.windows.net");

const result = await sendRequest(url, headers, body, method);
```

### What is the expected behavior?

The headers and body are returned with the response such as:

```bash
$ node sendDirectNotificationWithNode.js
Tracking ID: f5e8e6a8-bc8b-4a4b-b452-3e3c93299870
Correlation ID: f2340956-f677-447a-9d95-8eb2061e66b0
```

### What do you see instead?

```bash
$ bun run sendDirectNotificationWithNode.js
Sample: Error occurred:  939 |   }
940 |
941 |   _final(callback) {
942 |     this.#finished = true;
943 |
944 |     this.#fetchRequest = fetch(`${this.#protocol}//${this.#host}:${this.#port}${this.#path}`, {
                            ^
TypeError: fetch() request with GET/HEAD/OPTIONS method cannot have body.
```

### Additional information

This is blocking our usage of Bun for the Azure SDK as we are running into compatibility issues for the HTTPS stack

### PR DESCRIPTION
**Pull request by ThatOneBro**
Should potentially make progress on #2253, fixes checking for symbol which doesn't exist on Bun's `URL` object but does on Node's

### CODE REVIEW

File: src/bun.js/http.exports.js [hunk 1]
```diff
@@ -963,7 +963,7 @@ export class ClientRequest extends OutgoingMessage {
       body: this.#body,
       redirect: "manual",
       verbose: Boolean(process.env.BUN_JS_DEBUG),
```
Comment (Jarred-Sumner) [inline src/bun.js/http.exports.js]: can you hoist `process.env.BUN_JS_DEBUG` up to avoid the extra getter

File: src/bun.js/http.exports.js [hunk 2]
```diff
@@ -951,10 +951,10 @@ export class ClientRequest extends OutgoingMessage {
   _final(callback) {
     this.#finished = true;
     this[kAbortController] = new AbortController();
```
Comment (Jarred-Sumner) [inline src/bun.js/http.exports.js:955]: Since fetch supports `signal` now, we shouldn't need this anymore

Comment (Jarred-Sumner) [reply src/bun.js/http.exports.js:955]: oh this is for timeouts

I see

yeah we do need this

Comment (ThatOneBro) [reply src/bun.js/http.exports.js:955]: Yeah this is what Ciro just added

**Comment by Jarred-Sumner** [comment]
FWIW, I think the issue is from `body: ""`

we should hardcode `body` to `undefined` if the request method doesn't support it (such as GET or HEAD)

**Comment by ThatOneBro** [comment]
> FWIW, I think the issue is from `body: ""`
> 
> we should hardcode `body` to `undefined` if the request method doesn't support it (such as GET or HEAD)

Yeah the problem was that the original request was a `POST` request that was ending up as a `GET` request. The issue here is that it was likely parsed incorrectly due to the case where a `URL` object being passed resulted in erasure of the options object, and hence the method defaulting to `GET`

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**